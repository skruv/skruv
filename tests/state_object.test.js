/* global test expect */
import { createState, htmlFactory, render } from '../index.js'
const { body, div, html } = htmlFactory
self.SkruvWaitForAsync = true

const wait = time => new Promise(resolve => setTimeout(resolve, time))

test('update on array push', async () => {
  const sub = createState({ obj: {} })
  await render(html({}, body({},
    async function * () {
      for await (const state of sub) {
        yield Object.keys(state.obj).map(key => div({ 'data-key': key }, state.obj[key]))
      }
    }
  )))
  sub.obj.test = 'testvalue'
  sub.obj.test2 = 'testvalue2'
  await wait(20)
  expect(document.documentElement.innerHTML).toMatchSnapshot()
})
