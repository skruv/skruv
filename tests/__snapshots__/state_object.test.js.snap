// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update on array push 1`] = `"<body><div data-key=\\"test\\">testvalue</div><div data-key=\\"test2\\">testvalue2</div></body>"`;
